### PR TITLE
Remove generic meta class from resource and async resource classes

### PR DESCRIPTION
**Pull request by rmk135**

### CODE REVIEW

File: src/dependency_injector/resources.py [hunk 1]
```diff
@@ -31,7 +18,7 @@ def shutdown(self, resource: T) -> None:
         ...
 
 
-class AsyncResource(Generic[T], metaclass=ResourceMeta):
+class AsyncResource(Generic[T]):
```
Comment (EdwardBlair) [inline src/dependency_injector/resources.py:21]: Does this still not need to inherit from abc.ABC?

File: src/dependency_injector/resources.py [hunk 2]
```diff
@@ -1,26 +1,13 @@
...
 
 
-class ResourceMeta(GenericMeta, abc.ABCMeta):
-    def __getitem__(cls, item):
-        # Spike for Python 3.6
-        return cls(item)
-
-
-class Resource(Generic[T], metaclass=ResourceMeta):
+class Resource(Generic[T]):
```
Comment (EdwardBlair) [inline src/dependency_injector/resources.py:10]: Does this still not need to inherit from abc.ABC?

File: src/dependency_injector/resources.py (hunk 1 above)
Comment (rmk135) [reply src/dependency_injector/resources.py:21]: Hm... It probably needs to. I didn't have a test for unimplemented abc methods. I'll revise this later today. Thanks!

Comment (rmk135) [reply src/dependency_injector/resources.py:21]: Returned ``abc.ABCMeta`` inheritance and added tests in https://github.com/ets-labs/python-dependency-injector/commit/1163ac59d4443c1cfbe2c4391752c83512086b99. I should be more watchful. Thanks for pointing out the issue.

**Pull request merged**